nengo/networks/ensemblearray.py

```diff
@@ -89,6 +89,10 @@ def __init__(self, n_neurons, n_ensembles, ens_dimensions=1,
 
         super(EnsembleArray, self).__init__(label, seed, add_to_container)
 
+        for param in ens_kwargs:
+            if isinstance(ens_kwargs[param], np.ndarray):
+                ens_kwargs[param] = nengo.dists.Samples(ens_kwargs[param])
```

Comment:
Yeah, so like we talked about in the dev meeting, the issue there is that then we lose the model construction-time validation of shapes. This isn't a simple no-op, it means that if people are accessing `ensemble.encoders` they will never get a NumPy array, which is almost always what they want if they're accessing it to do some math or some such. I see the benefit of this distribution for fixing this specific problem, but I think it's premature to switch a lot of core functionality over to it because of issues like this. Not saying we won't, but I'd prefer it in a separate PR. I'm happy to switch over other specific instances in which nothing changes to the user, but yeah, mostly I wanted to get this out quickly and haven't looked around for other places to do this.